From cfc47d65674e4341912e8a0a6fc0072cd2e8e595 Mon Sep 17 00:00:00 2001
From: Evangelos Foutras <evangelos@foutrelis.com>
Date: Fri, 17 Sep 2021 20:49:24 +0300
Subject: [PATCH] Fix signal disconnect in thunar_window_unrealize

Commit c2a97b896f04 replaced the gtk_widget_queue_draw callback with
thunar_window_clipboard_manager_changed. The same change needs to be
made in thunar_window_unrealize, which was still referencing the old
function (gtk_widget_queue_draw).

Fixes: c2a97b896f04 ("Fix file-icon update when file is cut (Issue #588)")
Closes: https://gitlab.xfce.org/xfce/thunar/-/issues/653
---
 thunar/thunar-window.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/thunar/thunar-window.c b/thunar/thunar-window.c
index d464e338..27cd22f7 100644
--- a/thunar/thunar-window.c
+++ b/thunar/thunar-window.c
@@ -1820,7 +1820,7 @@ thunar_window_unrealize (GtkWidget *widget)
   ThunarWindow *window = THUNAR_WINDOW (widget);
 
   /* disconnect from the clipboard manager */
-  g_signal_handlers_disconnect_by_func (G_OBJECT (window->clipboard), gtk_widget_queue_draw, widget);
+  g_signal_handlers_disconnect_by_func (G_OBJECT (window->clipboard), thunar_window_clipboard_manager_changed, widget);
 
   /* let the GtkWidget class unrealize the window */
   (*GTK_WIDGET_CLASS (thunar_window_parent_class)->unrealize) (widget);
